sparse_ho/ho.py

```diff
@@ -54,26 +61,17 @@ def _grad_search(
         _get_val_grad, proj_hyperparam, log_alpha0, monitor,
         n_outer=100, verbose=False, tolerance_decrease='constant', tol=1e-5,
         t_max=10000):
-    """
-    This line-search code is taken from here:
-    https://github.com/fabianp/hoag/blob/master/hoag/hoag.py
-
-    Parameters
-    --------------
-    TODO
-
-    """
 
     try:
-        lambdak = log_alpha0.copy()
-        old_lambdak = lambdak.copy()
+        log_alphak = log_alpha0.copy()
```

Comment:
this try except can be improved IMO. It's not obvious than you're making an if based on if there's a single parameter alpha or a full array